src/services/userService.js

```diff
@@ -0,0 +1,44 @@
+const { data } = require('../data');
+
+const getAll = () => {
+  return data.users;
+};
+
+const getById = (id) => {
+  return data.users.find((user) => user.id === id) || null;
+};
+
+const updateUser = ({ id, name }) => {
+  const user = getById(id);
+
+  Object.assign(user, { name });
+
+  return user;
+};
+
+const createUser = (name) => {
+  const user = {
+    name,
+    id: data.nextId++,
+  };
+
+  data.users.push(user);
+
+  return user;
+};
```

Comment:
In the createUser function, you are directly incrementing the nextId property of the data object. This can lead to issues if multiple requests are processed at the same time. Consider using an atomic operation or a unique identifier generator to ensure uniqueness of ids.